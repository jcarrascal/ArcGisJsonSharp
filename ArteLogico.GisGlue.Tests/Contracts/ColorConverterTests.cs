using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ArteLogico.GisGlue.Contracts;
using Newtonsoft.Json;
using NUnit.Framework;
using System.IO;

namespace ArteLogico.GisGlue.Tests.Contracts
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ColorConverterTestsApprovals")]
    public class ColorConverterTests
    {
        [Test]
        public void RoundTripSerialization_WhenGivenAColor_ThenMatchesTheInput()
        {
            this.CheckRoundTripSerialization<Color>("ColorConverterTestsApprovals");
        }

        [Test]
        public void RoundTripSerialization_WhenGivenAColorContainer_ThenMatchesTheInput()
        {
            /*var cont = new ColorContainer { Background = new Color { R = 100 } };
            string result = JsonHelper.Serialize(cont, Formatting.Indented);
            System.Console.WriteLine(result);*/
            this.CheckRoundTripSerialization<ColorContainer>("ColorConverterTestsApprovals");
        }

        public class ColorContainer
        {
            public Color Background { get; set; }
        }
    }
}
