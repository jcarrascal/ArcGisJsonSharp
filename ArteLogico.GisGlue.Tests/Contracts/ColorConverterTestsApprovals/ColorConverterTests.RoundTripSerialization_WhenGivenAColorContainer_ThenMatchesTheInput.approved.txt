{
  "background": [
    19,
    19,
    19,
    255
  ]
}